File: QTi/backend/models/alert.py
from sqlalchemy import Column, Integer, String, Boolean, DateTime, JSON
from sqlalchemy.sql import func
from ..database import Base

class Alert(Base):
    __tablename__ = "alerts"

    id = Column(Integer, primary_key=True, index=True)
    type = Column(String, index=True)  # position_limit, balance, order, system
    exchange = Column(String, index=True)
    symbol = Column(String, index=True)
    message = Column(String)
    severity = Column(String)  # info, warning, error
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    read = Column(Boolean, default=False)

class AlertSettings(Base):
    __tablename__ = "alert_settings"

    id = Column(Integer, primary_key=True, index=True)
    position_limit_threshold = Column(Integer, default=10)  # Процент от баланса
    enabled_exchanges = Column(JSON, default=list)  # Список включенных бирж
    enabled_symbols = Column(JSON, default=list)  # Список включенных символов
    notification_channels = Column(JSON, default=dict)  # Настройки каналов уведомлений 

File: QTi/frontend/src/services/alert.ts
import { api } from './api';

export interface Alert {
    id: string;
    type: 'position_limit' | 'balance' | 'order' | 'system';
    exchange: string;
    symbol: string;
    message: string;
    severity: 'info' | 'warning' | 'error';
    created_at: string;
    read: boolean;
}

export interface AlertSettings {
    position_limit_threshold: number; // Процент от баланса
    enabled_exchanges: string[];
    enabled_symbols: string[];
    notification_channels: {
        email: boolean;
        telegram: boolean;
        web: boolean;
    };
}

class AlertService {
    async getAlerts(): Promise<Alert[]> {
        const response = await api.get('/alerts');
        return response.data;
    }

    async getUnreadAlerts(): Promise<Alert[]> {
        const response = await api.get('/alerts/unread');
        return response.data;
    }

    async markAlertAsRead(alertId: string): Promise<void> {
        await api.post(`/alerts/${alertId}/read`);
    }

    async markAllAlertsAsRead(): Promise<void> {
        await api.post('/alerts/read-all');
    }

    async getAlertSettings(): Promise<AlertSettings> {
        const response = await api.get('/alerts/settings');
        return response.data;
    }

    async updateAlertSettings(settings: Partial<AlertSettings>): Promise<AlertSettings> {
        const response = await api.put('/alerts/settings', settings);
        return response.data;
    }

    async deleteAlert(alertId: string): Promise<void> {
        await api.delete(`/alerts/${alertId}`);
    }

    async clearAllAlerts(): Promise<void> {
        await api.delete('/alerts');
    }
}

export const alertService = new AlertService(); 